internal/ignition/ignition.go

```diff
@@ -71,7 +72,7 @@ func (g *installerGenerator) UploadToS3(ctx context.Context, s3Client s3wrapper.
 }
 
 // Generate generates ignition files and applies modifications.
-func (g *installerGenerator) Generate(installConfig []byte) error {
+func (g *installerGenerator) Generate(ctx context.Context, s3Client s3wrapper.API, installConfig []byte) error {
```

Comment:
Why does it get s3 client and not init it in the main as part of `installerGenerator` configuration? 
Same comment for all `installerGenerator` functions.